From 43f8043dea23347206ef0d6d8c9f49c5989862c1 Mon Sep 17 00:00:00 2001
From: Arthur Liberman <arthur_liberman@hotmail.com>
Date: Sat, 15 Mar 2025 01:53:13 +0200
Subject: [PATCH 11/17] Revert "AMLUtils: drop support of 'disp_cap' and
 'disp_add'"

This reverts commit 435d1e379730ba6afe6c0ad41244bf0412112ba9.
---
 xbmc/utils/AMLUtils.cpp | 33 +++++++++++++++++++++++++++------
 1 file changed, 27 insertions(+), 6 deletions(-)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index 945943aa24..ed3f08a2ca 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -858,18 +858,39 @@ bool aml_probe_resolutions(std::vector<RESOLUTION_INFO> &resolutions)
 {
   std::string valstr, addstr;
 
-  valstr = aml_get_drmDevice_modes();
+  CSysfsPath user_dcapfile{CSpecialProtocol::TranslatePath("special://home/userdata/disp_cap")};
 
-  CSysfsPath vesa{"/flash/vesa.enable"};
-  if (vesa.Exists())
+  if (!user_dcapfile.Exists())
   {
-    CSysfsPath vesa_cap{"/sys/class/amhdmitx/amhdmitx0/vesa_cap"};
-    if (vesa_cap.Exists())
+    valstr = aml_get_drmDevice_modes();
+
+    CSysfsPath vesa{"/flash/vesa.enable"};
+    if (vesa.Exists())
+    {
+      CSysfsPath vesa_cap{"/sys/class/amhdmitx/amhdmitx0/vesa_cap"};
+      if (vesa_cap.Exists())
+      {
+        addstr = vesa_cap.Get<std::string>().value();
+        valstr += "\n" + addstr;
+      }
+    }
+
+    CSysfsPath custom_mode{"/sys/class/amhdmitx/amhdmitx0/custom_mode"};
+    if (custom_mode.Exists())
+    {
+      addstr = custom_mode.Get<std::string>().value();
+      valstr += "\n" + addstr;
+    }
+
+    CSysfsPath user_daddfile{CSpecialProtocol::TranslatePath("special://home/userdata/disp_add")};
+    if (user_daddfile.Exists())
     {
-      addstr = vesa_cap.Get<std::string>().value();
+      addstr = user_daddfile.Get<std::string>().value();
       valstr += "\n" + addstr;
     }
   }
+  else
+    valstr = user_dcapfile.Get<std::string>().value();
 
   if (aml_display_support_3d())
   {
-- 
2.48.1

